src/components/structural/header/SceneConfigMenu.js

```diff
@@ -81,18 +81,46 @@ class ConfigModal extends Component {
             open: false,
             skyColor: this.props.scene.settings.color,
             displaySkyColorPicker: false,
-            displayFloorColorPicker: false,
             anchorEl: null,
+            displayFloorColorPicker: false,
             qrCodeOpen: false,
             pwProtectOpen: false,
             shareOpen: false,
             addClassOpen: false,
             email: "",
             sendTo: [],
             collectionID: "",
-            value: "a"
+            value: "a",
         };
         this.emailRef = React.createRef();
+        
+        this.originalSettings = this.buildSettingsArr();
+
+        window.addEventListener("beforeunload", (event) => {
+            let finalSettings = this.buildSettingsArr();
+
+            if(!this.settingsEqual(finalSettings)){
+                event.preventDefault();
+                event.returnValue = "You may have unsaved changes!";
+            }
+        });
+    }
+
+    buildSettingsArr = () => {
+        const sceneSettings = this.props.scene.settings;
+
+        return [sceneSettings.canFly, sceneSettings.floorColor, 
```

Comment:
If you are just getting the values, I would try out `Object.values(scene.settings}` https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Object/values